Neos.Utility.Schema/Classes/TYPO3/Flow/Utility/SchemaValidator.php

```diff
@@ -624,4 +625,45 @@ protected function isDictionary(array $phpArray)
     {
         return array_keys($phpArray) !== range(0, count($phpArray) - 1);
     }
+
+    /**
+     * Generate a short description for the given schema-type to be used in error-messages
+     *
+     * @param mixed $type
+     * @return string
+     */
+    protected function getSchemaTypeInfo($type) {
+        if (is_scalar($type)) {
+            return (string)$type;
+        } else if (is_array($type)) {
+            if ($this->isSchema($type) && array_key_exists('type', $type)) {
+                return $this->getSchemaTypeInfo($type['type']);
+            } else if ($this->isNumericallyIndexedArray($type)) {
+                $subTypeInfos = [];
+                foreach ($type as $subType) {
+                    $subTypeInfos[] = $this->getSchemaTypeInfo($subType);
+                }
+                return '[' . implode(', ', $subTypeInfos) . ']';
+            } else {
+                return 'dictionary';
+            }
+        }
+        return gettype($type);
+    }
+
+    /**
+     * Generate a short description for the given value to be used in error-messages
+     *
+     * @param mixed $value
+     * @return string
+     */
+    protected function getValueInfo($value) {
+        if (is_scalar($value)) {
+            return (string)$value;
```

Comment:
Ok, so if **$value** is a scalar the actual value is returned as string, while otherwise **$value**'s type is returned. That's not easily expected from the method name or the method description.

Following the previous usages, there should actually be two distinct methods, one which outputs the type information for $value (everywhere it previously was `'type=' . gettype($value)`), and one which outputs the serialized value of **$value** (print_r/var_export), no?
The type information method would probably only check if **$value** is a dictionary type array and else just return `gettype($value)`.